### ISSUE TITLE
Gradle - Support npm dependencies in Kotlin JS and multiplatform

### ISSUE THREAD

**Issue by hfhbd** [renovatebot/renovate#19920]
### What would you like Renovate to be able to do?

Kotlin supports JS as a target with javascript dependencies from npm:
```kotlin
// build.gradle.kts
plugins {
    kotlin("js")
}

dependencies {
    implementation(npm("sql.js", "1.7.0"))
    implementation(npm(name = "sql.js", version = "1.7.0")) // notation with parameter names

    implementation(npm("sql.js", "1.7.0", true)) // notation with unrelated kotlin parameter
    implementation(npm(name = "sql.js", version = "1.7.0", generateExternal = true)) // notation with unrelated kotlin parameter

    implementation(devNpm("copy-webpack-plugin", "~9.1.0")) // dev scope with "normal" npm version notation
    implementation(peerNpm("copy-webpack-plugin", "~9.1.0")) // peer scope
}
```
Like always in kotlin: all parameter names are optionally.

### If you have any ideas on how this should be implemented, please tell us here.

Only a very general idea:
"Just" extract the npm dependency, run the normal npm update check and update the npm version variable in the Gradle script.
Maybe creating a temp npm file could work too.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
Does any lock file need to be updated afterwards, or just that `build.gradle.kts` ?

**Comment by hfhbd** [comment]
Yes: this lock file is stored by default under `kotlin-js-store/yarn.lock`.

But this lock file also needs to be updated when other js dependencies are updated, the Kotlin plugin contains some js dependencies too, like karma for unit tests or webpack for packing the js app. So when updating Kotlin to a newer version, you need to update this lock file too.
There is a Gradle task to update this lock file: `./gradlew kotlinUpgradeYarnLock` because you could change the default location of the lock file.
You (the user) could also update the built-in dependencies (like karma/webpack) with some configs in Gradle.
And keep in mind, you can also add an `npm` dependency from another Gradle plugin, so the final lock file could not be the same when updating it without Gradle due to new/conflicting npm version constraints.

So in the first step, I would not update this lock file. Or only if there is a file in the default location.

For me (!) creating a PR notifying me "here is a new npm version but you need to run `./gradlew kotlinUpgradeYarnLock` by yourself" would be enough for the first step.

**Comment by Churro** [comment]
I agree with @zharinov that it would be best to leverage the existing parsing capabilities.

`gradle` manager is currently tailored to the `MavenDatasource` only. It may need some minor adjustments to widen compatibility for additional ecosystems. I think I could provide this piece of abstraction over the next few days. Extracting these deps should then be easily doable by just adding further patterns.
